app/models/proposal.rb

```diff
@@ -30,6 +30,8 @@ class Proposal < ActiveRecord::Base
 
   acts_as_taggable
 
+  # we do not have "dependent:" declaration on "steps" because there can be circular references
```

Comment:
I do think that the difference between doing this in the DB vs with AR is relevant. 

I can imagine my future self adding `dependent: :destroy` and removing the DB constraint (silly future me!) and not realizing the side effects of doing that. 

Would it be possible to write a unit test for proposals that destroys one and confirms that all of its steps were destroyed? (ideally, in a way that `dependent: :destroy` would not do so that we have a failing test in the case that future me ever tries to make the change mentioned above)